cassiopeia/preprocess/pipeline.py

```diff
@@ -34,6 +35,130 @@
 progress = tqdm
 
 
+class PreprocessError(Exception):
+    pass
+
+
+def convert_fastqs_to_unmapped_bam(
+    fastq_fps: List[str],
+    chemistry: Literal["dropseq", "10xv2", "10xv3", "slideseq2"],
+    output_directory: str,
+    name: Optional[str] = None,
+    n_threads: int = 1,
+) -> str:
+    """Converts FASTQs into an unmapped BAM based on a chemistry.
+
+    This function converts a set of FASTQs into an unmapped BAM with appropriate
+    BAM tags.
+
+    Args:
+        fastq_fps: List of paths to FASTQ files. Usually, this argument contains
+            two FASTQs, where the first contains the barcode and UMI sequences
+            and the second contains cDNA. The FASTQs may be gzipped.
+        chemistry: Sample-prep/sequencing chemistry used. The following
+            chemistries are supported:
+            * dropseq: Droplet-based scRNA-seq chemistry described in
```

Comment:
Sorry - one more chemistry that would be good to support is inDrop, I think. Does `ngs` know how to deal with this chemistry?

Reply:
Done